src/renderers/shared/shared/__tests__/refs-test.js

```diff
@@ -289,6 +289,28 @@ describe('ref swapping', () => {
     const a = ReactTestUtils.renderIntoDocument(<A />);
     expect(a.refs[1].nodeName).toBe('DIV');
   });
+
+  it('supports updates with unmounts correctly when using refs', () => {
+    const ReactDOM = require('ReactDOM');
+    const container = document.createElement('div');
+    const log = [];
+    const ref = element => {
+      const result = element == null ? null : element.nodeName.toLowerCase();
+      log.push(result);
+      return result;
+    };
+    class Foo extends React.Component {
+      render() {
+        return this.props.useDiv
+          ? <div ref={ref}/>
+          : <span ref={ref} />;
+      }
+    }
+    ReactDOM.render(<Foo useDiv={true} />, container);
+    ReactDOM.render(<Foo useDiv={false} />, container);
+    expect(log).toEqual(['div', null, 'span']);
```

Comment:
oh whoops, didn’t mean to actually commit this. This was testing the behavior difference for refs between the test-renderer and DOM renderer re https://github.com/facebook/react/pull/8628#issuecomment-271478813